src/components/Table/ProductsTable.vue

```diff
@@ -9,51 +9,86 @@
       hide-details
     ></v-text-field>
   </v-card-title>
-  <v-data-table
-    :headers="tableColumns"
-    :items="productsInTable"
-    :search="search"
-  >
+  <v-data-table :headers="tableColumns" :items="allProducts" :search="search">
+    <template v-slot:item.resourceContent="{ item }">
+      <v-icon @click="openDialog(item, 'resources')">mdi-cube</v-icon>
+    </template>
+
+    <template v-slot:item.productsContent="{ item }">
+      <v-icon @click="openDialog(item, 'products')">mdi-cube-outline</v-icon>
+    </template>
+
     <template v-for="(_, slot) in $slots" v-slot:[slot]="scope">
       <slot :name="slot" v-bind="scope || {}" />
     </template>
   </v-data-table>
+
+  <resource-content-dialog
+    v-if="isResourceDialogOpen"
+    v-model="isResourceDialogOpen"
+    :data="resourceDialogData"
+    @close-dialog="closeDialog('resources')"
+  ></resource-content-dialog>
+
+  <products-content-dialog
+    v-if="isProductsDialogOpen"
+    v-model="isProductsDialogOpen"
+    :data="productsDialogData"
+    @close-dialog="closeDialog('products')"
+  >
+  </products-content-dialog>
 </template>
 
 <script setup>
-import { ref, computed, inject } from "vue";
+import { ref, computed, watch, toRefs } from "vue";
 import { VDataTable } from "vuetify/labs/VDataTable";
 import { useStore } from "vuex";
-const snackbarProvider = inject("snackbarProvider");
 
-const { products, additionalColumns } = defineProps({
+const props = defineProps({
   products: Array,
-  additionalColumns: Array,
+  additionalColumnsLeft: Array,
+  additionalColumnsRight: Array,
 });
 
+const { products, additionalColumnsLeft, additionalColumnsRight } =
+  toRefs(props);
+
 const tableColumns = computed(() =>
-  additionalColumns
-    ? [...additionalColumns, ...store.getters["products/getColumns"]]
-    : store.getters["products/getColumns"]
+  additionalColumnsLeft.value
+    ? [...additionalColumnsLeft.value, ...store.getters["products/getColumns"]]
+    : additionalColumnsRight.value
+    ? [...store.getters["products/getColumns"], ...additionalColumnsRight.value]
+    : [...store.getters["products/getColumns"]]
```

Comment:
```suggestion
  [
    ...(additionalColumnsLeft.value ?? []),
    ...store.getters["products/getColumns"],
    ...(additionalColumnsRight.value ?? []),
  ]
```

Reply:
dont understand here

Reply:
I am suggesting writing the logic for the columns included in a different way. The 2 ternary operators are very hard to read. What I propose looks more readable. Basically if the additionalColumns are passed they'd be included.

Reply:
updated